### ISSUE TITLE
make lighter weight, maybe

### ISSUE THREAD

**Issue by cdaringe** [dino-dna/checkup#11]
# problem

electron is dope, but everybody looovvvves to rail on it because it's like bringing a bazooka to a knife fight.

node-qt isn't really viable.  https://github.com/revery-ui/revery/issues/322 _is_, but w/out tray support, not yet.

even with all that said, JS for checks is >> better than some crummy DSL, and we can DI great dependencies for our users.  embedding lua wont be an option.  luas cool, but doesn't outweigh ^.  in the event that revery becomes viable, we may still embed node, or deno, or opt out of scripting altogether.  maybe!

**Comment by swashcap** [comment]
Yeah…I thought about a [macOS-only menu bar extra](https://developer.apple.com/design/human-interface-guidelines/macos/extensions/menu-bar-extras/). If we still with Apple's recommend approach we'd need some UX changes.

JS/TS is a pretty great scripting language. We could maybe adopt Sketch's approach (unsure how they implement, maybe just JSCore?): https://developer.sketch.com/plugins/javascript-api
